### DISCUSSION TITLE
Subclassing and member overridability

### DISCUSSION THREAD

**Discussion by chriskrycho** [typed-ember/semver#10] [General]
While reading and thinking about emberjs/rfcs#819, I realized we have a small gap: the spec currently notes that whether a class is intended to be subclassed or not is part of its private API:

> **Documentation of subclassibility:** Exported classes may be defined by documentation to be user-subclassible or not.

However, it’s also reasonable for someone to document that a given method is public to *call* but not public to *override*—which is also related to the points made on #4 about user-constructibility and user-consumability. We need to think about how to teach this, especially given that at present there is no TS equivalent of `final` (and it doesn't seem likely there will be any time soon!).

**Comment by tjjfvi** [comment]
> which is also related to the points made on https://github.com/typed-ember/semver/discussions/4 about user-constructibility and user-consumability

Not only is it related; I believe that it is a special case of user-constructibility – overriding a method is analogous to constructing a function of the type of the method, and so with a concept of user-constructibility, I think non-overiddability would follow (though it would still be worth speaking to directly, for clarity)
